collagraph/renderers/pyside/__init__.py

```diff
@@ -0,0 +1,95 @@
+from PySide6 import QtCore, QtGui, QtWidgets
+from PySide6.QtWidgets import (
+    QBoxLayout,
+    QCheckBox,
+    QComboBox,
+    QDialogButtonBox,
+    QGroupBox,
+    QLabel,
+    QLineEdit,
+    QMainWindow,
+    QMenuBar,
+    QPushButton,
+    QRadioButton,
+    QSlider,
+    QSpinBox,
+    QStatusBar,
+    QTextEdit,
+    QTreeView,
+    QWidget,
+)
+
+
+# Pre-populated cache for types
+TYPE_MAPPING = {
+    "Button": QPushButton,
+    "CheckBox": QCheckBox,
+    "ComboBox": QComboBox,
+    "Label": QLabel,
+    "LineEdit": QLineEdit,
+    "MenuBar": QMenuBar,
+    "RadioButton": QRadioButton,
+    "DialogButtonBox": QDialogButtonBox,
+    "GroupBox": QGroupBox,
+    "Slider": QSlider,
+    "SpinBox": QSpinBox,
+    "StatusBar": QStatusBar,
+    "TextEdit": QTextEdit,
+    "TreeView": QTreeView,
+    "Widget": QWidget,
+    "Window": QMainWindow,
+    # Layout directions
+    "TopToBottom": QBoxLayout.Direction.TopToBottom,
+    "LeftToRight": QBoxLayout.Direction.LeftToRight,
+    "RightToLeft": QBoxLayout.Direction.RightToLeft,
+    "BottomToTop": QBoxLayout.Direction.BottomToTop,
+}
+
+
+def name_to_type(name, modules=None, orig=None):
+    # IDEA: use a dict as cache (or lru_cache). Might speed things up a bit?
+    # Using a dict might be handy, because we can specify certain types in advance?
```

Comment:
Even if you use lru cache (which you definitely should) you can still keep this lookup and have some predefined mappings :)